cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/NewCommand.java

```diff
@@ -165,6 +165,16 @@ public void execute() {
             return;
         }
 
+        if (!ProjectUtils.validateInitialNumericsOfName(packageName)) {
+            CommandUtil.printError(errStream,
+                    "invalid package name : '" + packageName + "' :\n" +
+                            "Package name cannot have initial numeric characters.",
+                    null,
+                    false);
+            CommandUtil.exitError(this.exitWhenFinish);
```

Comment:
We should not fail the project creation. We should replace the disallowed characters and continue after printing a message. See L178-L180

Reply:
For package name validation. the only occasion that should fail is when the user has specified the package name in Ballerina.toml. If the Ballerina.toml is empty, then we should properly derive it to match the allowed pattern.

Reply:
Shouldn't that case also should fail ? `a` is not a good default IMO. We can inform the user that directory name is not a valid package name and they can either the change the directory name or add a valid package name in Ballerina.toml.

Reply:
Just tested and realized that we fail the package creation when there is an invalid initial character. We can do the same for digits as well.

Reply:
@hevayo should we do the same for init command as well? Since we produce an empty Ballerina.toml, should we still fail the command saying the package name is invalid?